src/Microsoft.Identity.Web.GraphServiceClient/GraphServiceCollectionExtensions.cs

```diff
@@ -103,14 +103,13 @@ private static GraphServiceClient CreateGraphServiceClient(IServiceProvider serv
         {
             var authorizationHeaderProvider = serviceProvider.GetRequiredService<IAuthorizationHeaderProvider>();
             var options = serviceProvider.GetRequiredService<IOptions<GraphServiceClientOptions>>();
-            var httpClientFactory = serviceProvider.GetRequiredService<IHttpClientFactory>();
             var microsoftGraphOptions = options.Value;
             if (microsoftGraphOptions.Scopes == null)
             {
                 Throws.ArgumentNullException("scopes", IDWebErrorMessage.CalledApiScopesAreNull);
             }
 
-            var httpClient = httpClientFactory.CreateClient("GraphServiceClient");
+            var httpClient = GraphClientFactory.Create();
```

Comment:
I think it's a breaking change for partners who would have provided their HttpClientFactory.
@jennyf19 this means we should not take this in rel/v2 and we should document it,

Reply:
Maybe check if there is a factory for "GraphServiceClient" honor it, and otherwise?

Reply:
In the TokenAcquisition constructor:

https://github.com/AzureAD/microsoft-identity-web/blob/77df2a63d759beb5bbeb9c1260c8efdffd9fff94/src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquisition.cs#L101

use the locator pattern to get a IMsalAspNetCoreHttpClientFactory if there is one in the service provider.

 _httpClientFactory = serviceProvider.GetService<IMsalAspNetCoreHttpClientFactory> ?? new MsalAspNetCoreHttpClientFactory(httpClientFactory);

Reply:
From what I can tell, this extension only adds the token acquisition service and GraphServiceClient if they are not already present, or if their lifetime does not match.
This code would not be executed in the case where the partners have configured their own.

The situation where an HTTPClient has been configured under the name of "GraphServiceClient" is one I am not sure if it can be checked without resulting in a default HTTPClient being created.

Any further thoughts are highly appreciated @jmprieur @jennyf19.